### PR TITLE
Create hyperparameter-optimization-Maxie-M.py

### PR DESCRIPTION
**Pull request by Maxie-M**
I wasn't sure what you meant by your last change request. In my original paper I feel it was clearly discussed, what the space of hyperparameters are being considered. Along with this the very next sentence states the utilization of RandomSearchCV for hyperparameter optimization which in turn was how I did the nested resampling. Although I did go ahead and add greater detail to both of these points and also corrected some grammatical mistakes I found while adding on to my paper. 

From my Original Paper:
"Two machine learning algorithms were chosen for the model training and performing the hyperparameter optimization. The algorithms selected were the Decision Tree Classifier and Random Forest Classifier. For this a search space was created for hyperparameters, this included: 
* Minimum Depth                       * Minimum Samples Split  
* Minimum Features                   * Minimum Samples Leaf 
On top of this RandomSearchCV has been utilized to perform hyperparameter optimization...." 

Fixed Paper: 
"Two machine learning algorithms were chosen for the model training and performing the hyperparameter optimization. The algorithms selected were the Decision Tree Classifier and Random Forest Classifier. For this a search space was created for hyperparameters, this included:  
* Maximum Depth                 * Minimum Samples Split 
* Maximum Features             *  Minimum Samples Leaf 
For both the Decision Tree Classifier and the Random Forest Classifier. The Random Forest Classifier also will be considering n_estimators, meaning it will consider the number of trees in the forest. On top of this RandomSearchCV has been utilized to perform nested  resampling for hyperparameter optimization. The  way this works is like double-checking where data is split into multiple sets of training and testing. Now for each of these splits it will be further split for validation in finding the best parameters. Helping assess models accuracy without overfitting...."

If you have further corrections or requests please let me know, thank you!

### CODE REVIEW

**Comment by larskotthoff** [comment]
Sorry for being unclear -- your description should contain enough detail to reproduce what you did. Please describe the ranges of the hyperparameter values that you tuned, i.e. what was the minimum and maximum number of estimators? Similarly, how many CV folds did you use in RandomSearchCV?

And why does performance get *worse* after optimization?
